lib/irc/Scheduler.js

```diff
@@ -0,0 +1,107 @@
+/*eslint no-invalid-this: 0*/
+"use strict"
+var promiseutil = require("../promiseutil");
+var Promise = require("bluebird");
+var logging = require("../logging");
+var log = logging.get("scheduler");
+
+function ScheduledQueue (server) {
```

Comment:
But the important difference is with ```_consume``` using ```setTimeout```. If I use ```Queue```, we can't do any extra delay in the critical section (line 27) and we can't do a ```setTimeout``` to retry after the reconnection interval. I could inherit from ```Queue```, but I'd have to overload the consumption method anyway.